src/promptflow-evals/promptflow/evals/evaluate/_eval_run.py

```diff
@@ -282,42 +320,74 @@ def log_artifact(self, artifact_folder: str) -> None:
                 local_file_path = os.path.join(root, f)
                 local_paths.append(local_file_path)
-        # Now we need to reserve the space for files in the artifact store.
-        headers = {
-            "Content-Type": "application/json",
-            "Accept": "application/json",
-            "Content-Length": str(len(json.dumps(remote_paths))),
-            "x-ms-client-request-id": str(uuid.uuid1()),
-        }
+
+        # We will write the artifacts to the workspaceblobstore
+        datastore = self._ml_client.datastores.get_default(include_secrets=True)
+        account_url = f"{datastore.account_name}.blob.{datastore.endpoint}"
+        svc_client = BlobServiceClient(
+            account_url=account_url, credential=self._get_datastore_credential(datastore))
+        for local, remote in zip(local_paths, remote_paths['paths']):
+            blob_client = svc_client.get_blob_client(
+                container=datastore.container_name,
+                blob=remote['path'])
+            with open(local, 'rb') as fp:
+                blob_client.upload_blob(fp, overwrite=True)
+
+        # To show artifact in UI we will need to register it. If it is a promptflow run,
+        # we are rewriting already registered artifact and need to skip this step.
+        if self._is_promptflow_run:
+            return
+        url = (
+            f"https://{self._url_base}/artifact/v2.0/subscriptions/{self._subscription_id}"
+            f"/resourceGroups/{self._resource_group_name}/providers/"
+            f"Microsoft.MachineLearningServices/workspaces/{self._workspace_name}/artifacts/register"
+        )
+
         response = self.request_with_retry(
-            url=self.get_artifacts_uri(), method="POST", json_dict=remote_paths, headers=headers
+            url=url,
+            method="POST",
+            json_dict={
+                "origin": "ExperimentRun",
+                "container": f"dcid.{self.info.run_id}",
+                "path": EvalRun.EVALUATION_ARTIFACT,
+                "dataPath": {
+                    "dataStoreName": datastore.name,
+                    "relativePath": posixpath.join(root_upload_path, EvalRun.EVALUATION_ARTIFACT),
+                },
+            },
         )
         if response.status_code != 200:
-            self._log_error("allocate Blob for the artifact", response)
-            return
-        empty_artifacts = response.json()["artifactContentInformation"]
-        # The response from Azure contains the URL with SAS, that allows to upload file to the
-        # artifact store.
-        for local, remote in zip(local_paths, remote_paths["paths"]):
-            artifact_loc = empty_artifacts[remote["path"]]
-            blob_client = BlobClient.from_blob_url(artifact_loc["contentUri"], max_single_put_size=32 * 1024 * 1024)
-            with open(local, "rb") as fp:
-                blob_client.upload_blob(fp)
+            self._log_error('register artifact', response)
+

```

Comment:
Can we reuse what is in v2 SDK ?

Reply:
What kind of mechanism do we want to use? [OpsLogger](https://github.com/Azure/azure-sdk-for-python/blob/aa710350672e4db762d9892a3e53d174503e89fd/sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_logger_utils.py#L22)?

Reply:
In this case we also will need lazy imports of V2 SDK.